hyperjumptech/monika
# Refactor: Convert TODO Comments Into Issues

# Monika Pull Request (PR)

## What feature/issue does this PR add

- Convert a TODO comment into an issue #1260. 
- Make the hardcoded notification sender changeable through a parameter.
- Separate sending error notification for the TLS checker function due to poor abstraction.
- Remove irrelevant TODO comments.

It will resolve #1010.

## How to test

Run the following configuration. You should receive a TLS error notification (Please set your own notification):

```yaml
probes:
  - id: example
    requests:
      - url: 'https://example.com'
certificate:
  domains:
    - example.com
    - expired.badssl.com
  reminder: 30
```

### Before
<img width="606" alt="Screenshot 2024-03-26 at 12 02 17 PM" src="https://github.com/hyperjumptech/monika/assets/15191978/f3c813d6-b06b-440f-b706-8b8f89c512a1">

### After (unchanged)
<img width="596" alt="Screenshot 2024-03-26 at 12 03 12 PM" src="https://github.com/hyperjumptech/monika/assets/15191978/b8fdc159-8ee9-4990-8c2c-8b9a2fd8ff09">
